prism_compile.c

```diff
@@ -2193,8 +2193,10 @@ pm_compile_node(rb_iseq_t *iseq, const pm_node_t *node, LINK_ANCHOR *const ret,
       case PM_NIL_NODE:
         PM_PUTNIL_UNLESS_POPPED
         return;
-      case PM_NO_KEYWORDS_PARAMETER_NODE:
+      case PM_NO_KEYWORDS_PARAMETER_NODE: {
+        ISEQ_BODY(iseq)->param.flags.accepts_no_kwarg = TRUE;
```

Comment:
Got this from [here](https://github.com/jemmaissroff/ruby/blob/5ded3fa417ef7e3b9509e1876072df636150cfb8/compile.c#L2079-L2081). It's still showing the wrong message, `wrong number of arguments (given 1, expected 0)` instead of `no keywords accepted`. But I'm fairly sure that's a bug due to the fact that we haven't implemented `KeywordRestParameterNode`, since `def foo(**args); end; foo(a: 1)` currently gives the same error (incorrectly)